File: Nutriologia/Modulos/Funciones.vb
Imports System.Data.SqlClient
Module Funciones
    Public ActiveUser As New Integer
    Public cmd As New SqlCommand
    Public dr As SqlDataReader
    Public query As String
    Public numID As Integer

    Public Sub GenerarNumID(textbox, Tabla)
        cmd.Connection = conn
        cmd.CommandType = CommandType.Text

        numID = -1
        Try
            query = "SELECT * FROM " & Tabla
            cmd.CommandText = query
        Catch ex As Exception
            MsgBox("Error en la consulta")
        End Try

        Try
            dr = cmd.ExecuteReader

            If dr.HasRows Then
                While dr.Read()
                    'Obtener el numero del ID del ultimo registro
                    numID = Val(dr(0).ToString)
                End While
            End If

            dr.Close()

            If (numID <= 0) Then
                numID = 1
            Else
                numID = numID + 1
            End If

            textbox.Text = numID
        Catch ex As Exception

        End Try
    End Sub

    'Funcion para mostrar una tabla en un dataGridView
    Public Sub CargaDatos(Tabla, DataGridView)
        Dim oda As New SqlDataAdapter
        Dim ods As New DataSet
        Dim datos As String
        datos = ""

        If Len(datos) = 0 Then
            query = "SELECT * FROM " & Tabla
            oda = New SqlDataAdapter(query, conn)  'Ejecutar la query (consulta) en la BD a la que apunta conn
            ods.Tables.Add(Tabla)         'Generar tabla en memoria
            oda.Fill(ods.Tables(Tabla))   'Poblar tabla
            DataGridView.DataSource = ods.Tables(Tabla)
            DataGridView.Visible = True
        End If
    End Sub

    Public Sub CargaDataProcedure(Instruccion, NumID, Tabla, DataGridView)
        Dim oda As New SqlDataAdapter
        Dim ods As New DataSet
        Dim datos As String
        datos = ""

        If Len(datos) = 0 Then
            query = Instruccion & " " & NumID
            oda = New SqlDataAdapter(query, conn)  'Ejecutar la query (consulta) en la BD a la que apunta conn
            ods.Tables.Add(Tabla)         'Generar tabla en memoria
            oda.Fill(ods.Tables(Tabla))   'Poblar tabla
            DataGridView.DataSource = ods.Tables(Tabla)
            DataGridView.Visible = True
        End If
    End Sub

    Public Sub IdentificarUsuario(DataGridView, Nombre)
        Try
            query = "SELECT * FROM Usuario WHERE Nombre = '" & Nombre & "'"
            Dim oda As New SqlDataAdapter
            Dim ods As New DataSet
            oda = New SqlDataAdapter(query, conn)  'Ejecutar la query (consulta) en la BD a la que apunta conn
            ods.Tables.Add("Usuario")         'Generar tabla en memoria
            oda.Fill(ods.Tables("Usuario"))   'Poblar tabla
            DataGridView.DataSource = ods.Tables("Usuario")
        Catch ex As Exception

        End Try
    End Sub
End Module

File: C# Nutriologia/Clases/Funciones.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Windows.Forms;

namespace C__Nutriologia.Clases
{
    public class Funciones: ConexionBD
    {
        public static int ActiveUser = new int();
        public static SqlCommand cmd = new SqlCommand();
        public static SqlDataReader dr;
        public static string query;
        public static int numID;

        public static void GenerarNumID(TextBox textbox, DataTable Tabla)
        {
            cmd.Connection = conn;
            cmd.CommandType = CommandType.Text;

            numID = -1;
            try
            {
                query = "SELECT * FROM " + Tabla;
                cmd.CommandText = query;
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Error en la consulta");
            }

            try
            {
                dr = cmd.ExecuteReader();

                if (dr.HasRows)
                {
                    while (dr.Read())
                        // Obtener el numero del ID del ultimo registro
                        numID = (int)Conversion.Val(dr[0].ToString());
                }

                dr.Close();

                if ((numID <= 0))
                    numID = 1;
                else
                    numID = numID + 1;

                textbox.Text = numID.ToString();
            }
            catch (Exception ex)
            {
            }
        }

        // Funcion para mostrar una tabla en un dataGridView
        public static void CargaDatos(DataTable Tabla, DataGrid DataGridView)
        {
            SqlDataAdapter oda = new SqlDataAdapter();
            DataSet ods = new DataSet();
            string datos;
            datos = "";

            if (Strings.Len(datos) == 0)
            {
                query = "SELECT * FROM " + Tabla;
                oda = new SqlDataAdapter(query, conn);  // Ejecutar la query (consulta) en la BD a la que apunta conn
                ods.Tables.Add(Tabla);         // Generar tabla en memoria
                oda.Fill(ods.Tables(Tabla));   // Poblar tabla
                DataGridView.DataSource = ods.Tables(Tabla);
                DataGridView.Visible = true;
            }
        }

        public static void CargaDataProcedure(object Instruccion, object NumID, object Tabla, object DataGridView)
        {
            SqlDataAdapter oda = new SqlDataAdapter();
            DataSet ods = new DataSet();
            string datos;
            datos = "";

            if (Strings.Len(datos) == 0)
            {
                query = Instruccion + " " + NumID;
                oda = new SqlDataAdapter(query, conn);  // Ejecutar la query (consulta) en la BD a la que apunta conn
                ods.Tables.Add(Tabla);         // Generar tabla en memoria
                oda.Fill(ods.Tables(Tabla));   // Poblar tabla
                DataGridView.DataSource = ods.Tables(Tabla);
                DataGridView.Visible = true;
            }
        }

        public static void IdentificarUsuario(object DataGridView, object Nombre)
        {
            try
            {
                query = "SELECT * FROM Usuario WHERE Nombre = '" + Nombre + "'";
                SqlDataAdapter oda = new SqlDataAdapter();
                DataSet ods = new DataSet();
                oda = new SqlDataAdapter(query, conn);  // Ejecutar la query (consulta) en la BD a la que apunta conn
                ods.Tables.Add("Usuario");         // Generar tabla en memoria
                oda.Fill(ods.Tables("Usuario"));   // Poblar tabla
                DataGridView.DataSource = ods.Tables("Usuario");
            }
            catch (Exception ex)
            {
            }
        }
    }
}
